cpp/autosar/src/rules/A0-1-3/UnusedLocalFunction.ql

```diff
@@ -19,6 +19,44 @@ import codingstandards.cpp.autosar
 import codingstandards.cpp.DynamicCallGraph
 import codingstandards.cpp.deadcode.UnusedFunctions
 
+/** Checks if a function call exists to the function
+ * passed in the arguments.
+ */
+predicate isCalled(Function unusedFunction) {
+  exists (FunctionCall f | unusedFunction.getACallToThisFunction() = f)
+}
+
+/** Checks if an overloaded function of
+ * the function passed in the arguments, is called.
+ */
+predicate overloadedFunctionIsCalled(Function unusedFunction) {
+  exists (Function f | f = unusedFunction.getAnOverload*() and isCalled(f))
+  or
+  unusedFunction.getNamespace().isAnonymous() and
+    exists (TopLevelFunction overloadedFunction |
+      overloadedFunction != unusedFunction and
+      ((overloadedFunction.getName() = unusedFunction.getName()) or
+        (overloadedFunction.getQualifiedName() =
+          unusedFunction.getQualifiedName()))
+      )
+}
+
+/** Checks if a Function is part of an unevaluated context. */
+predicate partOfUnevalutedContexts(Function unusedFunction) {
```

Comment:
I believe this case should be handled already. This is because even in unevaluated contexts, there still exists a `FunctionCall` which has the `Function` as a target, which should already be covered by the query. Can you try removing this predicate and see if the test case still passes?

Reply:
Yes, it passes. I'm removing this predicate. Thanks for your insights!